src/provider.rs

```diff
@@ -142,6 +145,48 @@ impl std::fmt::Debug for SeafowlTable {
     }
 }
 
+impl SeafowlTable {
+    // This code is partially taken from ListingTable but adapted to use an arbitrary
+    // list of Parquet URLs rather than all files in a given directory.
+    pub async fn partition_scan_plan(
+        &self,
+        projection: &Option<Vec<usize>>,
+        partitions: Vec<SeafowlPartition>,
+        filters: &[Expr],
+        limit: Option<usize>,
+        object_store: Arc<dyn ObjectStore>,
+    ) -> Result<Arc<dyn ExecutionPlan>> {
+        // Build a list of lists of PartitionedFile groups (one file = one partition for the scan)
+        let partitioned_file_lists: Vec<Vec<PartitionedFile>> =
+            future::try_join_all(partitions.iter().map(|p| async {
+                let path = Path::parse(&p.object_storage_id)?;
+                let meta = object_store.head(&path).await?;
+                Ok(vec![PartitionedFile {
+                    object_meta: meta,
+                    partition_values: vec![],
+                    range: None,
+                    extensions: None,
+                }]) as Result<_>
+            }))
+            .await?;
+
+        let config = FileScanConfig {
+            object_store_url: internal_object_store_url(),
+            file_schema: self.schema(),
+            file_groups: partitioned_file_lists,
+            statistics: Statistics::default(),
+            projection: projection.clone(),
+            limit,
+            table_partition_cols: vec![],
+        };
+
+        let format = ParquetFormat::default();
+        // TODO: filters here probably does nothing, since we handle pruning explicitly ourselves via
+        // `partitions` param
```

Comment:
Will it prune partitions twice, doing extra work? The `filters` might be used in the `EXPLAIN` output too (though I guess it'd be inconsistent with the list of partitions, since those are already pruned)

Reply:
Actually it will try to [prune row groups](https://github.com/apache/arrow-datafusion/blob/d7c0e420ce6362a253e1f55a873a0732052e0e2f/datafusion/core/src/physical_plan/file_format/parquet.rs#L424) in a single parquet file.

It is [also used](https://github.com/apache/arrow-datafusion/blob/d7c0e420ce6362a253e1f55a873a0732052e0e2f/datafusion/core/src/physical_plan/file_format/parquet.rs#L323-L331) for `EXPLAIN`.

So perhaps it is better to keep it there.